test/formal/combine_byte_shl.py

```diff
@@ -0,0 +1,24 @@
+from rule import Rule
+from opcodes import *
+
+"""
+byte(i, shl(256 - 8 * i, X))
+->
+and(X, 0xff)
+"""
+
+for i in range(32):
+	print(i)
+	rule = Rule()
+
+	n_bits = 256
+
+	# Input vars
+	X = BitVec('X', n_bits)
+
+	# Non optimized result
+	nonopt = BYTE(i, SHL(256 - 8 * i, X))
```

Comment:
```suggestion
	nonopt = BYTE(i, SHL(256 - 8 * (i + 1), X))
```
That's why the proof doesn't check out :-).